pkg/controllers/machine/disruption/drift_test.go

```diff
@@ -127,7 +127,212 @@ var _ = Describe("Drift", func() {
+					v1.LabelOSStable:           string(v1.Linux),
+				},
+				true,
+			),
+			Entry(
+				"should not return drifted if a node requirement is expanded",
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+				},
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux), string(v1.Windows)}},
+				},
+				map[string]string{
+					v1alpha5.LabelCapacityType: v1alpha5.CapacityTypeOnDemand,
+					v1.LabelOSStable:           string(v1.Linux),
+				},
+				false,
+			),
+			Entry(
+				"should not return drifted if a node requirement set to Exists",
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+				},
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpExists, Values: []string{}},
+				},
+				map[string]string{
+					v1alpha5.LabelCapacityType: v1alpha5.CapacityTypeOnDemand,
+					v1.LabelOSStable:           string(v1.Linux),
+				},
+				false,
+			),
+			Entry(
+				"should return drifted if a node requirement set to DoesNotExists",
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+				},
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpDoesNotExist, Values: []string{}},
+				},
+				map[string]string{
+					v1alpha5.LabelCapacityType: v1alpha5.CapacityTypeOnDemand,
+					v1.LabelOSStable:           string(v1.Linux),
+				},
+				true,
+			),
+			Entry(
+				"should not return drifted if a machine is grater then node requirement",
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelInstanceType, Operator: v1.NodeSelectorOpGt, Values: []string{"2"}},
+				},
+				[]v1.NodeSelectorRequirement{
+					{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+					{Key: v1.LabelInstanceType, Operator: v1.NodeSelectorOpGt, Values: []string{"10"}},
```

Comment:
nit: probably doesn't make a difference since it's a unit test, but consider using a non-stringed label. Like use LabelInstanceCPU instead of instanceType.

Reply:
Non-stringed isn't available to use in core.